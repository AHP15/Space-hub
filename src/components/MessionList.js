import MessionRow from './MessionRow';
import '../styles/MessionList.css';

// eslint-disable-next-line arrow-body-style
const MessionList = () => {
  return (
    <table className="mession_table">
      <thead>
        <tr className="table_head">
          <th>Mission</th>
          <th>Description</th>
          <th>Status</th>
          <th>{}</th>
        </tr>
      </thead>

      <tbody>
        {
          Array(10).fill(null).map((_, i) => (
            <MessionRow
              key={Math.random()}
              missionName="messtion name"
              description="Lorem ipsum dolor sit amet consectetur adipisicing elit. Tenetur delectus commodi ducimus sapiente voluptatibus illum, qui soluta quis perferendis inventore placeat culpa illo! Deserunt ducimus ab suscipit sint distinctio unde assumenda molestiae ipsum illum non reiciendis nesciunt exercitationem, quae quo quaerat nihil? Nihil dolore corrupti facilis nobis, vero cumque recusandae."
              changeBg={i % 2 === 0}
            />
          ))
        }
      </tbody>
    </table>
  );
};

export default MessionList;
