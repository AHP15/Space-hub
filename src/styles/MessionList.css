.mession_table {
  margin: 20px 70px;
  border: 1px solid lightgray;
  border-radius: 5px;
}

.table_head th,
.table_head td {
  padding: 20px;
  text-align: start;
  border: 1px solid lightgray;
}

.btn_row button {
  font-size: 14px;
  cursor: pointer;
  border: 1px solid grey;
  border-radius: 5px;
  text-align: center;
  padding: 5px 10px;
  color: grey;
  background-color: transparent;
}

.btn_row .not_member {
  background-color: grey;
  color: white;
}

.btn_row .member {
  background-color: rgb(52, 129, 154);
  color: white;
  border: 1px solid rgb(52, 129, 154);
}

.btn_row .joined {
  color: red;
  border: 1px solid red;
}
