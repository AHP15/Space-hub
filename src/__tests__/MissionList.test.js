import renderer from 'react-test-renderer';
// import { render, screen } from '@testing-library/react';
// import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';

import MessionList from '../components/MessionList';
import Store from '../slices/store';

it('renders NavBar correctly', () => {
  const tree = renderer.create(<Provider store={Store}><MessionList /></Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});
