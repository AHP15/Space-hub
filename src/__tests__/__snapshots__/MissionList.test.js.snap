// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders NavBar correctly 1`] = `
<table
  className="mession_table"
>
  <thead>
    <tr
      className="table_head"
    >
      <th>
        Mission
      </th>
      <th>
        Description
      </th>
      <th>
        Status
      </th>
      <th />
    </tr>
  </thead>
  <tbody>
    <tr
      className="table_head"
      style={
        Object {
          "backgroundColor": "rgba(211, 211, 211, 0.5)",
        }
      }
    >
      <th>
        mission name
      </th>
      <td>
        mission description
      </td>
      <td
        className="btn_row"
      >
        <button
          className="not_member"
          title="isMember"
          type="button"
        >
          NOT A MEMBER
        </button>
      </td>
      <td
        className="btn_row"
      >
        <button
          className=""
          onClick={[Function]}
          type="button"
        >
          Join Mission
        </button>
      </td>
    </tr>
    <tr
      className="table_head"
      style={
        Object {
          "backgroundColor": "",
        }
      }
    >
      <th>
        mission name
      </th>
      <td>
        mission description
      </td>
      <td
        className="btn_row"
      >
        <button
          className="member"
          title="isMember"
          type="button"
        >
          Active Member
        </button>
      </td>
      <td
        className="btn_row"
      >
        <button
          className="joined"
          onClick={[Function]}
          type="button"
        >
          Leave Mission
        </button>
      </td>
    </tr>
  </tbody>
</table>
`;
